src/services/transitGatewayAttachment/data.ts

```diff
@@ -84,6 +87,67 @@ const listTransitGatewayAttachmentsData = async ({
     )
   })
 
+const listTransitGatewayVpcAttachmentsData = async ({
+  ec2,
+  region,
+  nextToken: NextToken = '',
+}: {
+  ec2: EC2
+  region: string
+  nextToken?: string
+}): Promise<(TransitGatewayVpcAttachment & { region: string })[]> =>
+  new Promise<(TransitGatewayVpcAttachment & { region: string })[]>(resolve => {
+    let transitGatewayVpcAttachmentData: (TransitGatewayVpcAttachment & {
+      region: string
+    })[] = []
+    const transitGatewayVpcAttachmentList: TransitGatewayVpcAttachmentList = []
+    let args: DescribeTransitGatewayVpcAttachmentsRequest = {}
+
+    if (NextToken) {
+      args = { ...args, NextToken }
+    }
+
+    ec2.describeTransitGatewayVpcAttachments(
+      args,
+      (err: AWSError, data: DescribeTransitGatewayVpcAttachmentsResult) => {
+        if (err) {
+          errorLog.generateAwsErrorLog({
+            functionName: 'ec2:describeTransitGatewayVpcAttachments',
+            err,
+          })
+        }
+
+        if (!isEmpty(data)) {
```

Comment:
Would this promise never resolve if the data is empty?

Reply:
Yes, the promise is resolved in both cases (whether the data is empty or not) on line 146